src/utils/print.jl

```diff
@@ -124,6 +124,59 @@ function create_components_df(components::Components)
     return DataFrames.DataFrame(rows)
 end
 
+
+function Base.show(io::IO, ::MIME"text/plain", period::Union{Dates.TimePeriod, Dates.DatePeriod})
+    period = time_period_conversion(period)
+
+    milli_weeks = period - (period % Dates.Millisecond(604800000))
+    weeks = convert(Dates.Week, milli_weeks)
+    period = period - milli_weeks
+
+    milli_days = period - (period % Dates.Millisecond(86400000))
+    days = convert(Dates.Day, milli_days)
+    period = period - milli_days
+
+    milli_hours = period - (period % Dates.Millisecond(3600000))
+    hours = convert(Dates.Hour, milli_hours)
+    period = period - milli_hours
+
+    milli_minutes = period - (period % Dates.Millisecond(60000))
+    minutes = convert(Dates.Minute, milli_minutes)
+    period = period - milli_minutes
+
+    seconds = period - (period % Dates.Millisecond(1000)) # finding the seconds
+    seconds = convert(Dates.Second, seconds)
+    remainder = period % Dates.Millisecond(1000) #finding the remainding milliseconds
+    println(io, "$(weeks + days + hours + minutes + seconds + remainder)")
```

Comment:
Many people will be surprised by what this does if they only read it.  I suggest adding a comment saying that it will create a compound object and print that.